pkg/manager/logger/manager_test.go

```diff
@@ -122,8 +124,23 @@ func TestUpdateCfg(t *testing.T) {
 		ch <- &clonedCfg
 		ch <- &clonedCfg
 		test.action(lg)
-		logfiles := readLogFiles(t, dir)
-		test.check(logfiles)
+
+		// Backup files are removed by another goroutine, so there will be some delay.
+		// We check it multiple times until it succeeds.
+		ticker := time.NewTicker(3 * time.Second)
+		succeed := false
+		for !succeed {
+			select {
+			case <-ticker.C:
+				require.Fail(t, "time out", "%dth case", i)
```

Comment:
I think `Fail()` will continue the execution instead of failing right now. Probably use `Fatal` or `return` here.